objects/src/accounts/mod.rs

```diff
@@ -230,7 +233,51 @@ impl ToAdviceInputs for Account {
 // ================================================================================================
 
 #[cfg(feature = "serde")]
-mod serialization {
+mod vault_serialization {
```

Comment:
Why not define this directly on `AccountVault`? Given that `miden_utils::serde::Serializable` trait is implemented on `AccountVault`, and this delegates to that implementation, we might as well make this the `serde` implementation for the type?

Same applies to `AccountStorage` and `AccountCode`

Reply:
I view these as a temporary solution. What we should probably do is implement a custom `serde` serialization/deserializaiton for the `Account` which would serialize/deserialize using the same buffer (instead of allocating vectors for each component).

Reply:
Both could be true: `Account{Vault, Storage, Code}` would have a `serde::Serialize/Deserialization` based on `miden_utils::serde::Serializable`, but `Account` wouldn't use it for its own implementation of `serde::{De}Serialize`.

Ultimately I guess my question is: is there a good reason to explictly disallow people to {de}serialize `Account{Vault, Storage, Code}`? I currently don't see one